MapboxCoreNavigation/NavigationService.swift

```diff
@@ -234,14 +237,18 @@ public class MapboxNavigationService: NSObject, NavigationService, DefaultInterf
      - parameter eventsManagerType: An optional events manager type to use while tracking the route.
      - parameter simulationMode: The simulation mode desired.
      - parameter routerType: An optional router type to use for traversing the route.
+     - parameter skuIdentifier: An identifier corresponding to which billing model to use.
      */
     @objc required public init(route: Route,
                                directions: Directions? = nil,
                                locationSource: NavigationLocationManager? = nil,
                                eventsManagerType: NavigationEventsManager.Type? = nil,
                                simulating simulationMode: SimulationMode = .onPoorGPS,
-                               routerType: Router.Type? = nil)
+                               routerType: Router.Type? = nil,
+                               skuIdentifier: MBXAccountsSKUID? = .navigationSession)
     {
+        MBXAccounts.activateSKUID(skuIdentifier ?? .navigationSession)
```

Comment:
We could detect that the SKU ID hasn’t been activated yet and issue a one-time warning in the console to remind the developer to set it earlier. To differentiate between a Route that comes directly from the Directions API and a Route that’s deserialized from a file or built programmatically, we could check for one of the attributes added in mapbox/MapboxDirections.swift#342.

Reply:
This is no longer relevant. See https://github.com/mapbox/mapbox-navigation-ios/pull/2151#discussion_r294803296